### PR TITLE
update last checkin only if job is successful

### PR DESCRIPTION
**Pull request by mbussolotto**
## What does this PR change?

update last checkin only if job is successful

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: **add explanation. This can't be used if there is a GUI diff**
- No documentation needed: only internal and user invisible changes
- Documentation issue was created: [Link for SUSE Manager contributors](https://github.com/SUSE/spacewalk/issues/new?template=ISSUE_TEMPLATE_DOCUMENTATION.md&labels=documentation&projects=SUSE/spacewalk/31), [Link for community contributors](https://github.com/uyuni-project/uyuni-docs/issues/new).
- (OPTIONAL) [Documentation PR](https://github.com/uyuni-project/uyuni-docs/pulls)

- [x] **DONE**

## Test coverage
- [x] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/17252

- [x] **DONE**

## Changelogs

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by mbussolotto** [comment]
@mcalmer , what do you think about this change? I tested and it works, but IDK if it is the best approach or if it can cause any side effect...do you have any better idea to solve the issue? Thanks

**Comment by mbussolotto** [comment]
`executeSSHAction` in `SaltServerActionService` it's a taskomatic task and it correctly executed. `JobReturnEventMessageAction` is a spacewalk action, triggered by PGEventListener after any salt event is executed. So in our scenario, where a minion is down `executeSSHAction` fails but `JobReturnEventMessageAction` is executed as well, since a salt event has been executed. So yes, I think changes on `JobReturnEventMessageAction`  are enough. 
Adding @vzhestkov and @renner as reviewers, any other feedback is always appreciated :)

**Review by mcalmer — APPROVED** [review]

File: java/code/src/com/suse/manager/reactor/messaging/JobReturnEventMessageAction.java [hunk 1]
```diff
@@ -248,6 +248,11 @@ public void execute(EventMessage msg) {
                 jobReturnEvent.getMinionId());
         if (minion.isPresent()) {
             MinionServer m = minion.get();
+            if (jobResult.isEmpty()) {
+                LOG.error("Do not update server info since job=" +  jobReturnEvent.getJobId() + " in minion=" +
```
Comment (renner) [inline java/code/src/com/suse/manager/reactor/messaging/JobReturnEventMessageAction.java:252]: I am not sure if this should actually log as `error` here? Maybe for the context of not updating the server info, a log level of `info` could be enough, assuming that the actual failure of the job had been logged as `error` elsewhere.

Comment (mcalmer) [reply java/code/src/com/suse/manager/reactor/messaging/JobReturnEventMessageAction.java:252]: `info` is not logged by default. But maybe `warn` ?

Comment (mbussolotto) [reply java/code/src/com/suse/manager/reactor/messaging/JobReturnEventMessageAction.java]: changed in warn

**Pull request merged**